cumulus_library_data_metrics/utils.jinja

```diff
@@ -77,42 +77,44 @@
 {%- endmacro %}
 
 
--- Extracts the status field and returns a select with (id, status).
-{% macro extract_status(src) -%}
-(
-    -- Annoyingly, allergies and conditions use a codeableConcept field.
-    {% set systems = {
-        'AllergyIntolerance': 'http://terminology.hl7.org/CodeSystem/allergyintolerance-verification',
-        'Condition': 'http://terminology.hl7.org/CodeSystem/condition-ver-status',
-    } %}
-
-    -- Check what kind of resource we are dealing with
-    {% if src in systems %}
-        {% set join_table = src + '_extract_status_tmp_codes' %}
-        WITH {{ join_table }} AS {{ extract_codes(src, 'verificationstatus', system=systems[src]) }}
-    {%- endif %}
-
-    -- Now extract the status
-    SELECT
-        src.id,
-        {% if join_table %}
-        ARRAY_JOIN(ARRAY_SORT(join_table.codes), '; ')
+{% if db_type == "duckdb" -%}
+{% set transform = 'LIST_TRANSFORM' %}
+{% else %}
+{% set transform = 'TRANSFORM' %}
+{%- endif %}
```

Comment:
style convention nattering: i have been putting all sets like this at the top of the file, basically treating them like i would a constant. do you prefer the as needed positioning for this?

Reply:
Naw good point, moved up.